pytissueoptics/rayscattering/opencl/CLObjects.py

```diff
@@ -12,25 +12,37 @@
 
 
 class CLObject:
-    def __init__(self, name: str = None, struct: np.dtype = None):
+    def __init__(self, name: str = None, struct: np.dtype = None, autoReset: bool = False):
         self._name = name
         self._struct = struct
         self._declaration = None
         self._dtype = None
+        self._autoReset = autoReset
 
         self._HOST_buffer = None
         self._DEVICE_buffer = None
 
     def build(self, device: 'cl.Device', context):
+        self.make(device)
+        self._DEVICE_buffer = cl.Buffer(context, cl.mem_flags.READ_WRITE | cl.mem_flags.USE_HOST_PTR,
+                                        hostbuf=self._HOST_buffer)
+
+    def make(self, device):
         if self._struct:
             cl_struct, self._declaration = cl.tools.match_dtype_to_c_struct(device, self._name, self._struct)
             self._dtype = cl.tools.get_or_register_dtype(self._name, cl_struct)
-
         self._HOST_buffer = self._getHostBuffer()
-        self._DEVICE_buffer = cl.Buffer(context, cl.mem_flags.READ_WRITE | cl.mem_flags.USE_HOST_PTR,
-                                        hostbuf=self._HOST_buffer)
 
     def _getHostBuffer(self) -> np.ndarray:
+        if self._HOST_buffer is not None:
+            if self._autoReset:
+                self._initializeHostBuffer()
```

Comment:
This won't work. initializeHostBuffer only returns. Maybe rename to _initialHostBuffer(). Then properly set HostBuffer before returning array. (similar problem on line 43)